### PR TITLE
Change behavior of oneof numeric value field accessors

### PR DESCRIPTION
**Pull request by dmitryax**
The following methods are deprecated in favor of the new accessors that are more performant and require numeric type match:
  - `Exemplar.IntVal` -> `Exemplar.IntValue`
  - `Exemplar.DoubleVal` -> `Exemplar.DoubleValue`
  - `NumberDataPoint.IntVal` -> `NumberDataPoint.IntValue`
  - `NumberDataPoint.DoubleVal` -> `NumberDataPoint.DoubleValue`

Corresponding setter methods are also renamed for consistency without changing its logic:
  - `Exemplar.SetIntVal` -> `Exemplar.SetIntValue`
  - `Exemplar.SetDoubleVal` -> `Exemplar.SetDoubleValue`
  - `NumberDataPoint.SetIntVal` -> `NumberDataPoint.SetIntValue`
  - `NumberDataPoint.SetDoubleVal` -> `NumberDataPoint.SetDoubleValue`

Updates: https://github.com/open-telemetry/opentelemetry-collector/issues/4775

Similar to: https://github.com/open-telemetry/opentelemetry-collector/pull/4909

### CODE REVIEW

**Comment by dmitryax** [comment]
Patch coverage is below the target because of touching uncovered usage of the migrated methods.

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,18 @@
 - Add `pdata.AttributeMap.RemoveIf`, which is a more performant way to remove multiple keys (#4914)
 - Add `pipeline` key with pipeline identifier to processor loggers (#4968)
 
+### 🚩 Deprecations 🚩
+
+- Change behavior of oneof numeric value field accessors (#4950)
```
Comment (Aneurysm9) [inline CHANGELOG.md:18]: How does the behavior change?  Do these also panic if not carefully accessed?  That would be important to call out here.

Comment (dmitryax) [reply CHANGELOG.md:18]: Yes, the same behavior as the AttributeValue accessors. I'll update it once we agree on the API in https://github.com/open-telemetry/opentelemetry-collector/pull/4909

File: model/internal/cmd/pdatagen/internal/base_fields.go [hunk 2]
```diff
@@ -91,7 +91,14 @@ const copyToValueOneOfMessageTemplate = `	case ${typeName}:
...
+// Calling this function when ValueType() != ${valueType} will cause a panic.
 func (ms ${structName}) ${fieldName}() ${returnType} {
+	return (*ms.orig).Value.(*${originStructType}).${originFieldName}
+}
+
+// ${deprecatedFieldName} returns the ${lowerFieldName} associated with this ${structName}.
+// Deprecated: [v0.47.0] Use ${fieldName} instead.
+func (ms ${structName}) ${deprecatedFieldName}() ${returnType} {
 	return (*ms.orig).Get${originFieldName}()
 }
```
Comment (bogdandrutu) [inline model/internal/cmd/pdatagen/internal/base_fields.go:104]: I don't think we should generate this via the generator. You can write these manually.

**Comment by dmitryax** [comment]
Closing based on the decision made in https://github.com/open-telemetry/opentelemetry-collector/issues/4775

**Pull request closed without merging**